src/domain/app/notificationsContext/NotificationsContext.tsx

```diff
@@ -0,0 +1,74 @@
+import uniqueId from 'lodash/uniqueId';
+import React, {
+  createContext,
+  FC,
+  PropsWithChildren,
+  useMemo,
+  useState,
+} from 'react';
+
+import Notification, {
+  NotificationProps,
+} from '../../../common/components/notification/Notification';
+
+export type NotificationsContextProps = {
+  addNotification: (props: NotificationProps) => void;
+};
+
+export const NotificationsContext = createContext<
+  NotificationsContextProps | undefined
+>(undefined);
+
+export const NotificationsProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [notifications, setNotifications] = useState<NotificationProps[]>([]);
+
+  const value = useMemo<NotificationsContextProps>(() => {
+    return {
+      addNotification: (props: NotificationProps) =>
+        setNotifications((items) => [
+          ...items,
+          { ...props, id: uniqueId('notification-') },
+        ]),
+    };
+  }, [setNotifications]);
+
+  return (
+    <NotificationsContext.Provider value={value}>
+      {notifications.map((props, index) => {
+        const removeNotification = () =>
+          setNotifications((items) => items.filter((i) => i.id !== props.id));
+        const getNotificationStyle = (): React.CSSProperties => {
+          const offset = 32;
+          let topMargin = offset;
+
+          for (let i = 0; i < index; i = i + 1) {
```

Comment:
To me the for loop calculation here seems a bit ineffective so I was thinking that could something like this work better:
```
let topMargin = index > 0 ? index * 32 : 32;

if (index > 0) {
  const prevNotification = notifications[index - 1];
  const prevNotificationHeight =
    document.getElementById(prevNotification.id as string)?.children[0]
      ?.clientHeight ?? 0;
  topMargin += prevNotificationHeight;
}
```

Reply:
Actually NVM: my code doesn't work exactly like the for loop as it doesn't take into consideration all of the previous elements' heights. Let's call this The Big Friday Mistake.

The topMargin value would need to be remembered from the previous iteration round for my suggestion to otherwise work.

Reply:
If the need is to calculate notifications heights only once, we can do like this:

```
const heights = useMemo(
    () =>
      // Height of last added notification is 0 because it's not rendered yet.
      // In this case it doesn't matter because it's not used for top-margin calculation
      notifications.map(
        ({ id }) =>
          document.getElementById(id as string)?.children[0]?.clientHeight ?? 0
      ),
    [notifications]
  );
```

```
const getNotificationStyle = (): React.CSSProperties => {
          const offset = 32;
          const topMargin = heights
            .slice(0, -(notifications.length - index))
            .reduce((acc, curr) => acc + curr + offset, offset);

          return {
            top: topMargin,
            transform: 'translate3d(0px, 0px, 0px)',
            zIndex: 1000,
          };
        };
```

Implemented in https://github.com/City-of-Helsinki/linkedcomponents-ui/pull/239/commits/594251adfae2ac0654caa7d6f561972f6c8d245d